src/main/java/net/sf/jabref/model/entry/BibEntry.java

```diff
@@ -52,6 +53,14 @@
     public static final String DEFAULT_TYPE = "misc";
 
     private String id;
+
+    // This id is set by the remote database system (DBS).
+    // It has to be unique on remote DBS for all connected JabRef instances.
+    // The old id above does not satisfy this requirement.
+    private int sharedID;
+
+    private int version; // Needed for version controlling if used on shared database
```

Comment:
As both version and sharedID are used for the same thing, we should create a final immutable class containing both fields so that the relationship between them is shown clearly.